internal/vault/testutils_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// TestHTTPServer creates a test HTTP server that handles requests until
+// the listener returned is closed.
+// XXX: based off of github.com/hashicorp/vault/api/client_test.go
```

Comment:
What does `XXX` mean here? Is there a problem with this code?

Reply:
Not sure, fixed in 7fa55d5ffeda41e8c2705cc32de3385a04e0ce1e